kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/RecordEncryptionST.java

```diff
@@ -115,24 +129,141 @@ void produceAndConsumeMessage(String namespace, TestKmsFacade<?, ?, ?> testKmsFa
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When {} messages '{}' are sent to the topic '{}'", numberOfMessages, MESSAGE, topicName);
+        KroxyliciousSteps.produceMessages(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
+
+        LOGGER.info("Then the messages are consumed");
+        List<ConsumerRecord> resultEncrypted = KroxyliciousSteps.consumeMessageFromKafkaCluster(namespace, topicName, clusterName,
+                Constants.KAFKA_DEFAULT_NAMESPACE, numberOfMessages, Duration.ofMinutes(2));
+        LOGGER.info("Received: {}", resultEncrypted);
+
+        assertKekVersionWithinParcel(resultEncrypted, ":v1:", testKekManager);
+
+        LOGGER.info("When KEK is rotated");
+        testKekManager.rotateKek("KEK_" + topicName);
+
+        try {
+            Thread.sleep(5000);
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+
+        LOGGER.info("And {} messages '{}' are sent to the topic '{}'", numberOfMessages, MESSAGE, topicName);
+        KroxyliciousSteps.produceMessages(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
+
+        LOGGER.info("Then the messages are consumed");
+        List<ConsumerRecord> resultEncryptedRotatedKek = KroxyliciousSteps.consumeMessageFromKafkaCluster(namespace, topicName, clusterName,
+                Constants.KAFKA_DEFAULT_NAMESPACE, numberOfMessages, Duration.ofMinutes(2));
+        LOGGER.info("Received: {}", resultEncryptedRotatedKek);
+
+        List<ConsumerRecord> finalEncryptedResults = new ArrayList<>(resultEncryptedRotatedKek);
+        finalEncryptedResults.removeAll(resultEncrypted);
+        assertKekVersionWithinParcel(finalEncryptedResults, ":v2:", testKekManager);
+    }
+
+    private void assertKekVersionWithinParcel(List<ConsumerRecord> consumerRecords, String expectedValue, TestKekManager testKekManager) {
+        assertThat(consumerRecords)
+                .withFailMessage("expected messages not received! Consumer records is empty")
+                .isNotEmpty();
+
+        assertThat(testKekManager.getClass().getSimpleName().toLowerCase())
+                .withFailMessage("Another KMS different from Vault is not currently supported!")
+                .contains("vault");
+
+        assertThat(consumerRecords.stream())
+                .withFailMessage(expectedValue + " is not contained in the ciphertext blob!")
+                .allMatch(r -> r.getValue().contains(expectedValue));
+    }
+
+    @SuppressWarnings("java:S2925")
+    @TestTemplate
+    void produceAndConsumeMessageWithRotatedKEK(String namespace, TestKmsFacade<?, ?, ?> testKmsFacade) {
+        testKekManager = testKmsFacade.getTestKekManager();
+        testKekManager.generateKek("KEK_" + topicName);
+        int numberOfMessages = 1;
+        ExperimentalKmsConfig experimentalKmsConfig = new ExperimentalKmsConfig(5L, 5L, 5L, 5L);
```

Comment:
How do you know you are testing the configuration parameter you think you are?  Just set one and leave the others at their default values.

Reply:
Yes!  That was the point we were both making when @k-wall made the 5s suggestion.